File: springboot/src/main/java/com/edufelip/meer/security/guards/AppHeaderGuard.java
package com.edufelip.meer.security.guards;

import com.edufelip.meer.security.SecurityProperties;
import jakarta.servlet.http.HttpServletRequest;

public class AppHeaderGuard {
    public static final String APP_HEADER = "X-App-Package";
    private final SecurityProperties props;

    public AppHeaderGuard(SecurityProperties props) {
        this.props = props;
    }

    public void validate(HttpServletRequest request) {
        if (!props.isRequireAppHeader()) return;
        String header = request.getHeader(APP_HEADER);
        if (header == null || !header.equals(props.getAppPackage())) {
            throw new GuardException("Missing or invalid " + APP_HEADER + " header");
        }
    }
}

File: springboot/src/main/kotlin/com/edufelip/meer/security/guards/AppHeaderGuard.kt
package com.edufelip.meer.security.guards

import com.edufelip.meer.security.SecurityProperties
import jakarta.servlet.http.HttpServletRequest

class AppHeaderGuard(private val props: SecurityProperties) {
    fun validate(request: HttpServletRequest) {
        if (!props.requireAppHeader) return
        val header = request.getHeader(APP_HEADER)
        if (header == null || header != props.appPackage) {
            throw GuardException("Missing or invalid $APP_HEADER header")
        }
    }

    companion object {
        const val APP_HEADER = "X-App-Package"
    }
}
